internal/graphapi/control_clone.go

```diff
@@ -0,0 +1,110 @@
+package graphapi
+
+import (
+	"context"
+
+	"github.com/theopenlane/core/internal/ent/generated"
+	"github.com/theopenlane/core/internal/ent/generated/control"
+	"github.com/theopenlane/core/pkg/enums"
+)
+
+func (r *mutationResolver) cloneControls(ctx context.Context, existingControls []*generated.Control, programID *string, ignoreStandard bool) ([]*generated.Control, error) {
+	createdControlIDs := make([]string, len(existingControls))
+
+	for _, control := range existingControls {
+		mappedControlIDs := []string{}
+		mappedControls := control.Edges.MappedControls
+
+		for _, mc := range mappedControls {
+			mappedControlIDs = append(mappedControlIDs, mc.ID)
+		}
+
+		controlInput := generated.CreateControlInput{
+			Tags:                   control.Tags,
+			RefCode:                control.RefCode,
+			Description:            &control.Description,
+			Source:                 &control.Source,
+			ControlType:            &control.ControlType,
+			Category:               &control.Category,
+			CategoryID:             &control.CategoryID,
+			Subcategory:            &control.Subcategory,
+			MappedCategories:       control.MappedCategories,
+			AssessmentObjectives:   control.AssessmentObjectives,
+			ControlQuestions:       control.ControlQuestions,
+			ImplementationGuidance: control.ImplementationGuidance,
+			ExampleEvidence:        control.ExampleEvidence,
+			References:             control.References,
+			MappedControlIDs:       mappedControlIDs,
+			Status:                 &enums.ControlStatusPreparing,
+		}
+
+		if !ignoreStandard {
+			if control.StandardID != "" {
+				controlInput.StandardID = &control.StandardID
+			}
+		}
+
+		if programID != nil {
+			controlInput.ProgramIDs = []string{*programID}
+		}
+
+		res, err := withTransactionalMutation(ctx).Control.Create().SetInput(controlInput).Save(ctx)
+		if err != nil {
+			return nil, parseRequestError(err, action{action: ActionCreate, object: "control"})
```

Comment:
helper function should just return the error and let the resolver handle the parsing
```suggestion
			return nil, err
```